lib/shared/addon/components/cluster-driver/driver-googlegke/component.js

```diff
@@ -251,4 +482,197 @@ export default Component.extend(ClusterDriver, {
+      url:    '/meta/gkeNetworks',
+      method: 'POST',
+      data:   {
+        credentials: get(this, 'config.credential'),
+        projectId:   get(this, 'config.projectId'),
+        zone:        get(this, 'config.zone'),
+      }
+    }).then((xhr) => {
+      const out = xhr.body.items || [];
+
+      set(this, 'networks', out);
+
+      if (get(this, 'mode') === 'new') {
+        set(this, 'config.network', out[0] && out[0].name)
+      }
+
+      return out;
+    }).catch((xhr) => {
+      set(this, 'errors', [xhr.body.error]);
+
+      return reject();
+    });
+  },
+
+  fetchSubnetworks() {
+    const zone = get(this, 'config.zone')
+    const zoneSplit = zone.split('-')
+
+    return get(this, 'globalStore').rawRequest({
+      url:    '/meta/gkeSubnetworks',
+      method: 'POST',
+      data:   {
+        credentials: get(this, 'config.credential'),
+        projectId:   get(this, 'config.projectId'),
+        region:      `${ zoneSplit[0] }-${ zoneSplit[1] }`,
+      }
+    }).then((xhr) => {
+      const out = xhr.body.items || [];
+
+      set(this, 'subNetworks', out);
+
+      return out;
+    }).catch((xhr) => {
+      set(this, 'errors', [xhr.body.error]);
+
+      return reject();
+    });
+  },
+
+  fetchServiceAccounts() {
+    return get(this, 'globalStore').rawRequest({
+      url:    '/meta/gkeServiceAccounts',
+      method: 'POST',
+      data:   {
+        credentials: get(this, 'config.credential'),
+        projectId:   get(this, 'config.projectId'),
+        zone:        get(this, 'config.zone'),
+      }
+    }).then((xhr) => {
+      const out = xhr.body.accounts || [];
+
+      set(this, 'serviceAccounts', out);
+      const filter = out.filter((o) => o.displayName === 'Compute Engine default service account')
+
+      if (get(this, 'mode') === 'new') {
+        set(this, 'config.serviceAccount', filter[0] && filter[0].uniqueId)
+      }
+
+      return out;
+    }).catch((xhr) => {
+      set(this, 'errors', [xhr.body.error]);
+
+      return reject();
+    });
+  },
+
+  validate() {
+    const model = get(this, 'cluster');
+    const errors = model.validationErrors();
+    const { intl, config = {} } = this
+    const { minNodeCount, maxNodeCount } = config
+
+    if (maxNodeCount < minNodeCount) {
+      errors.pushObject(intl.t('clusterNew.googlegke.maxNodeCount.minError'))
+    }
+
+    if (!get(this, 'cluster.name')) {
+      errors.pushObject(intl.t('clusterNew.name.required'))
+    }
+
+    const taints = get(this, 'taints') || []
+
+    if (taints.length > 0) {
+      const filter = taints.filter((t) => !t.key || !t.value)
+
+      if (filter.length > 0) {
+        errors.pushObject(intl.t('clusterNew.googlegke.taints.required'))
+      }
+    }
+
+    set(this, 'errors', errors);
+
+    return errors.length === 0;
+  },
+
+  willSave() {
+    const config = get(this, 'config') || {}
+    const clone = get(this, 'globalStore').createRecord({ type: 'googleKubernetesEngineConfig' })
```

Comment:
Whats the reason you're reseting the config before the cluster has been saved? seems like if we error here then we'd reset to an empty config?